app-connector-client/test/test_integration.js

```diff
@@ -1,115 +1,189 @@
+#!/usr/bin/env node
+'use strict'
 
 var request = require('supertest');
-var server = require("../server/app")("test/varkes_config.json");
+const mock = require('../server/app')
+const express = require("express")
 const fs = require("fs")
 const path = require("path")
+const kyma = require("@varkes/example-kyma-mock")
 const serviceMetadata = path.resolve("test/service-metadata.json")
 
-if (process.env.NODE_ENV == "test") {
-    require("../server/keys").generatePrivateKey()
+const port = 10001 //! listen in different port
+const tokenURL = `http://localhost:${port}/connector/v1/applications/signingRequests/info?token=123`
+
+describe("should work", () => {
+    var kymaServer
+    var server
+    before(async () => { //* start kyma mock before tests
+
+        await deleteKeysFile()

```

Comment:
Either chain the promises or execute them even in parallel so that you have only one await